drivers/blueview_driver/src/BlueViewRosDriver.cpp

```diff
@@ -0,0 +1,128 @@
+
+		color_img.reset(new cv_bridge::CvImage());
+		color_img->encoding = "bgra8";
+        grayscale_img->header.frame_id = frame_id;
+		color_pub = image_transport.advertise("image_color", 1);
+	}
+	if (do_raw) {
+		ping_msg.reset(new sonar::BlueViewPing);
+		raw_pub = nh.advertise<sonar::BlueViewPing>("ranges", 5);
+	}
+
+	std::string params;
+	if (nh.getParam("file", params))
+		sonar.init(BlueViewSonar::ConnectionType::FILE, params);
+	else if (nh.getParam("device", params))
+		sonar.init(BlueViewSonar::ConnectionType::DEVICE, params);
+	else throw std::runtime_error("Can not connect: neither 'file' or 'device' param set");
+
+	// Set Ranges, meters
+	BVTSDK::Head& head  = sonar.getHead();
+	float range_lower, range_upper;
+	if (nh.getParam("range/start", range_lower)) head.SetStartRange(range_lower);
+	if (nh.getParam("range/stop",  range_upper)) head.SetStopRange(range_upper);
+
+	// Handle fluid type, enum string
+	std::string fluid_type;
+	if (nh.getParam("fluid_type", fluid_type)) {
+		if (fluid_type == "saltwater") head.SetFluidType(BVTSDK::FluidType::Saltwater);
+		else if (fluid_type == "freshwater") head.SetFluidType(BVTSDK::FluidType::Freshwater);
+		else if (fluid_type == "other") head.SetFluidType(BVTSDK::FluidType::Other);
+		else ROS_ERROR("Fluid type (%s) invalid, must be 'saltwater', 'freshwater', or 'other'", fluid_type.c_str());
+	}
+
+	// Set sound speed, m/s
+	int sound_speed;
+	if (nh.getParam("sound_speed", sound_speed)) head.SetSoundSpeed(sound_speed);
+	
+	// Set analog gain adjustment in dB
+	float gain;
+	if (nh.getParam("gain_adjustment", gain)) head.SetGainAdjustment(gain);
+	
+	// Set "time variable analog gain", in dB/meter,
+	float tvg;
+	if (nh.getParam("tvg_slope", tvg)) head.SetTVGSlope(tvg);
+	
+	// Set dynamic power managment
+	bool dynamic_power;
+	if (nh.getParam("dynamic_power_management", dynamic_power)) head.SetDynamicPowerManagement(dynamic_power);
+	
+	// Set ping interval
+	float ping_interval;
+	if (nh.getParam("ping_interval", ping_interval)) head.SetPingInterval(ping_interval);
+
+	// Start loop
+	double period_seconds;
+	nh.param<double>("period_seconds", period_seconds, 0.1);
+	timer = nh.createTimer(ros::Duration(period_seconds),std::bind(&BlueViewRosDriver::loop, this, std::placeholders::_1));
+}
+void BlueViewRosDriver::loop(const ros::TimerEvent&)
+{
+	sonar.getNextPing();
+
+	// Do images
+	if(do_grayscale || do_color)
+	{
+		sonar.generateImage();
+		if (do_grayscale)
+		{
+			sonar.getGrayscaleImage(grayscale_img->image);
+			grayscale_pub.publish(grayscale_img->toImageMsg());
+		}
+		if (do_color)
+		{
+			sonar.getColorImage(color_img->image);
+			color_pub.publish(color_img->toImageMsg());
+		}
+	}
+	if (do_raw)
+	{
+		ping_msg->header.frame_id = frame_id;
+		ping_msg->header.stamp = ros::Time::now();
+		sonar.getRanges(ping_msg->bearings, ping_msg->ranges, ping_msg->intensities);
+		raw_pub.publish(ping_msg);
+	}
+
+}
+
+int main(int argc, char** argv)
```

Comment:
May I suggest:

> int main(int argc, char** argv)
> try
> {
>     ....
> }
> catch (const BVTSDK::SdkException& err)
> {
>     ...
> }

This keeps proper indentation and is the right way to do it.